misk/src/main/kotlin/misk/digester/Windower.kt

```diff
@@ -0,0 +1,108 @@
+package misk.digester
+import java.util.Calendar
+/** Window represents a time range */
+class Window (private val start: Calendar, private val end: Calendar) {
+  /** Returns true if the given time t falls within the window's Start <= t < End. */
+  fun contains(t: Calendar): Boolean {
+    return !t.before(start) && t.before(end)
+  }
+  /** Returns true when the other window has the same start and end times. */
+  fun equals(other: Window): Boolean {
+    return start.time == other.start.time && end.time == other.end.time
+  }
+  /** Returns string representing the start and end time */
+  fun string(): String {
+    return "[${start.time}, ${end.time})"
+  }
+}
+/** Windower contains multiple windows with a specified duration */
+class Windower {
+  private var windowSize: Int = 0
+  val startSecs: MutableList<Int> = mutableListOf()
+  /**
+   * Creates a new Windower, which will create windows of windowSecs duration.
+   * The size of the window must be in the range of 1 to 60 seconds, and must divide 60 without any remainder:
+   * 1, 2, 3, 4, 5, 6, 10, 12, 15, 20, 30 second window sizes are permitted.
+   * Stagger defines how many windows will contain a single time; a value of 1 means windows never overlap.
+   */
+  constructor(windowSecs: Int, stagger: Int) {
+    require (windowSecs > 0 || windowSecs <= 60)  {
+      "windowSecs must be in the range of (0, 60]"
+    }
+    require (60 % windowSecs == 0) {
+      "60 % windowSecs must be 0, not ${60 % windowSecs}"
+    }
+    require(stagger > 0 && stagger <= windowSecs) {
+      "stagger must be >= 1 and <= windowSecs"
+    }
+    var start = 0
+    while (start < 60) {
+      startSecs.add(start)
+      for (i in 1..(stagger-1)) {
+        startSecs.add(start + i*windowSecs/stagger)
+      }
+      start += windowSecs
+    }
+    this.windowSize = windowSecs
+  }
+  /**
```

Comment:
Add a blank line between each method. I think the autoformatter will do this for you.